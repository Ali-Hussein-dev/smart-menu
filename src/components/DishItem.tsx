import { useState } from 'react'
import { Drawer } from 'antd'
import Price from './Price'
import { ImInfo } from 'react-icons/im'

export const DrawerTitle: React.FC<{ name: string }> = ({ name: cuisine }) => (
  <span className=" text-blueGray-600">
    <div className="flex items-start gap-x-2">
      <ImInfo className="mt-1 text-blueGray-400" />{' '}
      <div className="flex flex-col items-start justify-start">
        <span>Zusatzstoffe Info:</span>
        <span className="text-sm font-thin">{cuisine}</span>
      </div>
    </div>
  </span>
)
interface Props {
  name: string
  description: string
  price: number
}
//=======================
const DishItem: React.FC<Props> = ({ price, name, description }) => {
  // hooks
  const [visible, setVisible] = useState(false)
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div className="flex justify-between bg-blueGray-400">
      <div className="">
        <span className="font-semibold">{name}</span>
        <p className="">{description}</p>
      </div>
      <Price price={price} visible={visible} setVisible={setVisible} />
      <Drawer
        title={<DrawerTitle name={name} />}
        placement={'bottom'}
        closable={true}
        onClose={() => setVisible(false)}
        visible={visible}
        key={'bottom'}
        headerStyle={{
          backgroundColor: '#F8FAFC',
          color: '#334155',
        }}
        bodyStyle={{
          color: '#94A3B8',
        }}
      >
        <p data-testid="drawer">Keine Vorgabe!</p>
      </Drawer>
    </div>
  )
}
export default DishItem
