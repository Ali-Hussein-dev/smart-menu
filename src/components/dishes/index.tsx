export * from './Lunch'
export * from './DishItem'
