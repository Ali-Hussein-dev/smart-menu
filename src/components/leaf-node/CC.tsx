//=======================Conditional Component
export const CC: React.FC<{
  isTrue: any
}> = ({ isTrue, children }) => {
  return <>{isTrue ? children : null}</>
}
