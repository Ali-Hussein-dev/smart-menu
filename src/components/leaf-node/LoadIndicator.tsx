import { Center, Spinner } from '@chakra-ui/react'

export const LoadIndicator: React.FC = () => (
  <Center h="7" size="xl">
    <Spinner size="xl" />
  </Center>
)
