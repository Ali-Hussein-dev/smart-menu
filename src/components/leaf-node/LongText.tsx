import { Text } from '@chakra-ui/react'
import react from 'react'

//=======================
export const LongText: React.FC = ({ children }) => {
  // hooks
  const [lines, setLines] = react.useState<undefined | number[]>([2, 3, 5])
  //--------------------------------------
  // functions
  function handleClick() {
    typeof lines === 'undefined' ? setLines([2, 3, 4]) : setLines(undefined)
  }
  //--------------------------------------
  return (
    <div
      data-testid="text-expand"
      className="cursor-default text-blueGray-500"
      onClick={() => handleClick()}
    >
      <Text noOfLines={lines}>{children}</Text>
    </div>
  )
}
