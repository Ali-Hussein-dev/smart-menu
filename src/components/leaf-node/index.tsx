export * from './LongText'
export * from './SvgIcon'
export * from './Price'
export * from './FruitIcon'
export * from './LoadIndicator'
export * from './SliderChildren'
export * from './MenuTop'
export * from './CC'