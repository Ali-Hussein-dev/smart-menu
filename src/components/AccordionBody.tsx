import DishItem from './DishItem'

//--------------------------------------
const ExtraItem: React.FC<{
  extra: {
    list: { name: string; price: number; label: string }[]
    header4: string
  }
}> = ({ extra }) => {
  // hooks
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div className="p-2 rounded-sm text-blueGray-600 bg-amber-100">
      <div className="flex">
        <h3 className="text-sm font-semibold md:text-lg text-blueGray-500">
          {extra.header4}{' '}
        </h3>
      </div>
      {extra.list.map((o) => (
        <div
          key={o.name}
          className="flex items-start justify-between gap-x-1 md:text-lg"
        >
          <span className="mb-2 text-sm font-normal md:text-lg ">
            {o.name}{' '}
          </span>
          <div className="flex items-center gap-x-1">
            <span className="text-sm font-light tracking-tight md:text-lg">
              {o.label}{' '}
            </span>
            <span className="font-normal">{o.price.toFixed(2)} </span>
          </div>
        </div>
      ))}
    </div>
  )
}
//=======================
const AccordionBody: React.FC<{
  extra?: any
  ImgSrc: string
  list: any[]
  id: number
}> = ({ extra, list: list, ImgSrc, id }) => {
  // hooks
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div data-testid={id.toString()} className="px-2">
      <img className="" src={ImgSrc} />
      {list.map((o) => (
        <DishItem
          key={o.name}
          name={o.name}
          description={o.description}
          price={o.price}
        />
      ))}
      {extra && <ExtraItem extra={extra} />}
    </div>
  )
}
export default AccordionBody
