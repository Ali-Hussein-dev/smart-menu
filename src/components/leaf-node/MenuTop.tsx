import { Avatar, Divider, HStack } from '@chakra-ui/react'
import { CC } from './CC'
import { SvgIcon } from './SvgIcon'
interface Assests {
  [name: string]: string
}
//=======================
export const MenuTop: React.FC<{
  header: string
  H?: React.ElementType
  assest?: Assests
  Component?: React.ReactNode
}> = ({ header = '', H = 'h2', assest = {}, Component }) => {
  // hooks
  //--------------------------------------
  // functions
  if (Object.keys(assest).length > 1) {
    throw new Error('Assest must have only ONE property')
  }
  if (!assest.imgSrc && !Component && !assest.avatarSrc) {
    return (
      <CC isTrue={header}>
        <H className="text-lg font-semibold text-center text-blueGray-600 ">
          {header}
        </H>
      </CC>
    )
  }
  //--------------------------------------
  if (assest.imgSrc) {
    return (
      <div className="flex flex-col px-2 mb-3">
        <CC isTrue={assest.imgSrc}>
          <img
            src={assest.imgSrc}
            alt={header}
            className="mx-auto mt-2 mb-2 rounded-lg"
          />
        </CC>
        <CC isTrue={header}>
          <H className="text-lg font-semibold text-center text-blueGray-600 ">
            {header}
          </H>
        </CC>
        <Divider variant="dashed" />
      </div>
    )
  }
  return (
    <HStack spacing={'8px'}>
      <CC isTrue={Component}>
        <SvgIcon color="text-orange-700">{Component}</SvgIcon>
      </CC>
      <CC isTrue={assest.avatarSrc}>
        <Avatar src={assest.avatarSrc} />
      </CC>
      <H className="text-sm font-semibold break-all text-blueGray-600">{header}</H>
    </HStack>
  )
}
