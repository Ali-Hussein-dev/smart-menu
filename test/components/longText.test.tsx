import { LongText } from '../../src/components/leaf-node'
import { render, screen } from '@testing-library/react'
const p =
  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque iusto dolorem veritatis, sapiente modi possimus aspernatur doloremque eveniet autem ipsum necessitatibus eius voluptatibus accusantium aperiam officia perspiciatis amet est dolorum iste. Facere dolorem maiores asperiores.'
describe('LongText', () => {
  test('LongText matches snapshot', () => {
    const { container } = render(<LongText>{p}</LongText>)
    expect(container).toMatchSnapshot()
  })
  test('text should be expanded when clicked', () => {
    const { asFragment } = render(<LongText>{p}</LongText>)
    const expand = screen.getByTestId('text-expand')
    expand.click()
    expect(asFragment).toMatchSnapshot()
  })
  test('text should be collapsed when second time clicked', () => {
    const { asFragment } = render(<LongText>{p}</LongText>)
    const expand = screen.getByTestId('text-expand')
    expand.click()
    expand.click()
    expect(asFragment).toMatchSnapshot()
  })
})
