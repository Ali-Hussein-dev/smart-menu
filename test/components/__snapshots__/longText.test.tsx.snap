// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LongText LongText matches snapshot 1`] = `
<div>
  <div
    class="leading-5 break-words cursor-default text-blueGray-400"
    data-testid="text-expand"
  >
    <div
      class="chakra-text css-r4i3pr"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque iusto dolorem veritatis, sapiente modi possimus aspernatur doloremque eveniet autem ipsum necessitatibus eius voluptatibus accusantium aperiam officia perspiciatis amet est dolorum iste. Facere dolorem maiores asperiores.
    </div>
  </div>
</div>
`;

exports[`LongText text should be collapsed when second time clicked 1`] = `[Function]`;

exports[`LongText text should be expanded when clicked 1`] = `[Function]`;
